### PR TITLE
feat(contacts): add respond to request and pending contact menu options

### LINKED ISSUE #6251: In Profile menu show options for "Respond to Contact Request", "Contact Request Pending"

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #6251

Adds the "Respond to Contact Request" and "Contact Request Pending" options to the MessageContextMenu

Also fixes some small issues with contact verification where the state of the incoming request was not correct

This branch is on top of https://github.com/status-im/status-desktop/pull/6377

image

image

### CODE REVIEW

File: ui/imports/shared/views/chat/MessageContextMenuView.qml [hunk 1]
```diff
@@ -239,6 +252,28 @@ StatusPopupMenu {
...
+            qsTr("ID Request Pending....") :
+            qsTr("Respond to ID Request...")
+        icon.name: "checkmark-circle"
+        enabled: root.isProfile && !root.isMe && root.isContact
+                                && !root.isBlockedContact && !root.isTrusted
+                                && (root.hasReceivedVerificationRequestFrom
+                                    || root.isVerificationRequestSent)
+        onTriggered: {
+            if (hasReceivedVerificationRequestFrom) {
+                root.openProfileClicked(root.selectedUserPublicKey, "respondToPendingRequest")
```
Comment (Khushboo-dev-cpp) [inline ui/imports/shared/views/chat/MessageContextMenuView.qml:268]: can we make these states Enums instead? would be more efficient?

**Review by Khushboo-dev-cpp — COMMENTED** [review]
LGTM :) Just one small comment.

File: ui/imports/shared/views/chat/MessageContextMenuView.qml (hunk 1 above)
Comment (jrainville) [reply ui/imports/shared/views/chat/MessageContextMenuView.qml]: Good idea. Done

**Comment by Hbouaz** [comment]
@jrainville please ping me on a DM when available, I would like some clarification about the feature 
Thanks

File: src/app_service/service/contacts/service.nim [hunk 2]
```diff
@@ -604,8 +604,9 @@ QtObject:
         return self.receivedIdentityRequests[publicKey]
 
       let response = status_contacts.getVerificationRequestFrom(publicKey)
-      result = response.result.toVerificationRequest()
-      self.receivedIdentityRequests[publicKey] = result
+      if not response.result.isNil and response.result.kind == JObject:
+        result = response.result.toVerificationRequest()
+        self.receivedIdentityRequests[publicKey] = result
```
Comment (endulab) [inline src/app_service/service/contacts/service.nim:609]: This is a minor comment but I find "getVerificationRequestFrom" a bit misleading - it gets verReq from status_go but also sets it in receivedIdentityRequests, unlike "getVerificationRequestSentTo" which only gets.

**Review by endulab — APPROVED** [review]
I am not familiar very much with verification logic but I did not found anything suspicious ;P

**Review by iurimatias — APPROVED** [review]

**Pull request merged**